src/04-Fundamentals-2.mdx

```diff
@@ -1,4 +1,121 @@
 ---
 name: 04. Fundamentals 2 Components
 ---
+
+import screen from './assets/cta-standard.png' 
+import screenOpen from './assets/cta-open.png' 
+import carp from './assets/carp.jpg' 
+
 # Fundamentals II: Components
+
+*Level: Competent Carp* 
+
+<img src={carp} width="100%"/>
+
+After you built a static page in the last chapter and learned about layout, content styling and semantics, you are now ready to take it a step further. In this chapter, you will learn about component based architecture and how to build components using SASS, BEM and Javascript. This should give you the necessary tools and knowledge to build state-of-the-art biotope components later on, and that's why you're here, right? Right. Let's get into it.
+
+## Preparation
+
+In this section, you will get to know package managers (NPM / Yarn) and a taskrunner (gulp) and how to use them in our project workflow. You will be using a micro frontend framework, that was specifically designed for this task.  
+We won't go into much detail about most of these topics, but if you want to read up on them, we put together a dandy list of links in the resources at the end of this section.
+
+### Step 1: Install Node.js and NPM
+
+Node.js is a way to run Javascript code without the need to use a browser. It's mainly used to run servers. You won't need to use it a lot, so it's not mandatory to do any work with it for this chapter.  
+Node.js comes with NPM, its very own package manager. This tool is used by the majority of web developers to install packages, libraries and other tools. It's pretty much your bread and butter and it is very much obligatory that you learn how to use it. The first step is to download and install it from [the Node.js homepage](https://nodejs.org/en/).  
+To ensure the installation worked correctly, type the following command into your command line:  `node -v && npm -v`  
+If you don't get any errors, everything worked out as planned. Congrats!
+
+### Step 2: Install Yarn
+
+Yarn is essentially the same as NPM, just faster and a bit better in general. It completes the same tasks as NPM though, that's all you need to know for now. Go ahead and get it from [the Yarn homepage](https://yarnpkg.com/lang/en/docs/install/#windows-stable).  
+Try `yarn -v` to check if the installation worked.
+
+### Step 3: Install gulp
+
+gulp is a taskrunner, which does a lot of work for you like compile, build, serve and other stuff. To install it, run the command `npm install gulp-cli -g` and then `npm install gulp -D`. 
+> *That's how easy it is to install packages with NPM, how neat!*  
+
+Gulp works with a so called `gulpfile.js`, which defines which tasks gulp completes for you. `gulpfiles` are written in Node.js (so essentially JS), for this task, we will provide you with one.
```

Comment:
gulpfiles are written in javascript, so is Gulp. To execute that javascript on your computer we use NodeJS as a javascript runtime. NodeJS is using the same as the chrome browser btw.